packages/patterns/src/disposables/disposables-group.ts

```diff
@@ -10,11 +10,15 @@ export class DisposablesGroup {
         const _disposables = Array.from(this.disposables).reverse();
         this.disposables.clear();
         for (const [disposable, details] of _disposables) {
-            await timeout(
-                disposeOf(disposable),
-                details.timeout,
-                `Disposal timed out: "${details.name}" after ${details.timeout}ms`
-            );
+            try {
+                await timeout(
+                    disposeOf(disposable),
+                    details.timeout,
+                    `Disposal timed out: "${details.name}" after ${details.timeout}ms`,
+                );
+            } catch (e) {
+                throw new Error(`Disposal failed: "${details.name}"\n${(e as Error)?.stack ?? ''}`);
```

Comment:
ignore the timeout error?

Reply:
why would you like to ignore it and not fail the current phase?

Reply:
the error from the timeout already include the message. it will throw the timeout and the error. maybe it's fine but not clean